include/deal.II/base/tensor_function.templates.h

```diff
@@ -123,9 +123,8 @@ template <int rank, int dim, typename Number>
 typename TensorFunction<rank, dim, Number>::gradient_type
 ConstantTensorFunction<rank, dim, Number>::gradient(const Point<dim> &) const
 {
-  static const Tensor<rank + 1, dim, Number> zero{};
-
-  return zero;
+  // Return a zero (=default initialized) tensor
+  return gradient_type();
```

Comment:
You need the full name, but just 
```suggestion
  return {};
```
should work as well.